assembly/src/parsers/ast/ast.rs

```diff
@@ -0,0 +1,256 @@
+use vm_core::utils::collections::BTreeMap;
+
+use vm_core::Felt;
+
+// AST
+// ================================================================================================
+
+// This holds the list of instructions supported in a Miden program.
+// This instruction list is used to hold reference to the instruction, and future be
+// used for MAST generation.
+pub enum Instruction {
+    Assert,
+    AssertEq,
+    Assertz,
+    Add,
+    AddImm(Felt),
+    Sub,
+    SubImm(Felt),
+    Mul,
+    MulImm(Felt),
+    Div,
+    DivImm(Felt),
+    Neg,
+    Inv,
+    Pow2Checked,
+    Pow2Unchecked,
+    Not,
+    And,
+    Or,
+    Xor,
+    Eq,
+    EqImm(Felt),
+    Neq,
+    NeqImm(Felt),
+    Eqw,
+    Lt,
+    Lte,
+    Gt,
+    Gte,
+
+    // ----- u32 manipulation ---------------------------------------------------------------
+
+    U32Test,
+    U32TestW,
+    U32Assert,
+    U32AssertW,
+    U32Split,
+    U32Cast,
+    U32CheckedAdd,
+    U32CheckedAddImm(Felt),
```

Comment:
For u32 instructions, we could actually just use `u32` as the datatype of the immediate value.

Reply:
fixed